From 989293a764919d4fcc008a6496af5f7957838f75 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Tue, 23 May 2023 15:22:16 +0530
Subject: [PATCH] Fixed Sepolicy Build Issue.

Getting following build error-:
Found prohibited permission granted for isolated like types.
Please replace your allow statements that involve "-isolated_app" with "-isolated_app_all".

From Android U onwards, it is not allowing permission in isolated_app
instead use of isolated_app_all.

Tracked-On: OAM-106853
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 camera/ipu_common/cameraserver.te        | 2 +-
 codec2/msdk-codec2/codec2_hal_default.te | 2 +-
 drm-default/hal_drm_clearkey.te          | 2 +-
 graphics/mesa/appdomain.te               | 4 ++--
 graphics/mesa_acrn/appdomain.te          | 2 +-
 vendor/domain.te                         | 2 +-
 6 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/camera/ipu_common/cameraserver.te b/camera/ipu_common/cameraserver.te
index 1c61865..78cb027 100644
--- a/camera/ipu_common/cameraserver.te
+++ b/camera/ipu_common/cameraserver.te
@@ -36,6 +36,6 @@ allow cameraserver hal_graphics_allocator_default_tmpfs:file rw_file_perms;
 allow cameraserver gpu_device:dir r_dir_perms;
 
 # Allow cameraserver to use fd from app,gralloc,and ashmem HAL
-allow cameraserver { appdomain -isolated_app }:fd use;
+allow cameraserver { appdomain -isolated_app_all }:fd use;
 allow cameraserver surfaceflinger:fd use;
 allow cameraserver hal_allocator_server:fd use;
diff --git a/codec2/msdk-codec2/codec2_hal_default.te b/codec2/msdk-codec2/codec2_hal_default.te
index bc7b2aa..a86d6ee 100644
--- a/codec2/msdk-codec2/codec2_hal_default.te
+++ b/codec2/msdk-codec2/codec2_hal_default.te
@@ -32,7 +32,7 @@ allow codec2_hal_default hal_graphics_allocator_default:fd use;
 allow codec2_hal_default hal_graphics_composer_default:fd use;
 allow codec2_hal_default hal_camera:fd use;
 allow codec2_hal_default bufferhubd:fd use;
-allow codec2_hal_default { appdomain -isolated_app }:fd use;
+allow codec2_hal_default { appdomain -isolated_app_all }:fd use;
 
 allow codec2_hal_default gpu_device:dir r_dir_perms;
 allow codec2_hal_default gpu_device:chr_file rw_file_perms;
diff --git a/drm-default/hal_drm_clearkey.te b/drm-default/hal_drm_clearkey.te
index 1e31ffb..057449d 100644
--- a/drm-default/hal_drm_clearkey.te
+++ b/drm-default/hal_drm_clearkey.te
@@ -8,4 +8,4 @@ hal_server_domain(hal_drm_clearkey, hal_drm)
 
 vndbinder_use(hal_drm_clearkey);
 
-allow hal_drm_clearkey { appdomain -isolated_app }:fd use;
+allow hal_drm_clearkey { appdomain -isolated_app_all }:fd use;
diff --git a/graphics/mesa/appdomain.te b/graphics/mesa/appdomain.te
index a7c05d3..f192881 100644
--- a/graphics/mesa/appdomain.te
+++ b/graphics/mesa/appdomain.te
@@ -2,5 +2,5 @@
 #allow appdomain surfaceflinger_tmpfs:file { read write };
 allow appdomain hal_graphics_allocator_default_tmpfs:file { read write map };
 allow appdomain gpu_device:dir r_dir_perms;
-allow { appdomain -isolated_app } sysfs_app_readable:file r_file_perms;
-allow { appdomain -isolated_app } proc_graphics:file r_file_perms;
+allow { appdomain -isolated_app_all } sysfs_app_readable:file r_file_perms;
+allow { appdomain -isolated_app_all } proc_graphics:file r_file_perms;
diff --git a/graphics/mesa_acrn/appdomain.te b/graphics/mesa_acrn/appdomain.te
index 616abc2..1f214b1 100644
--- a/graphics/mesa_acrn/appdomain.te
+++ b/graphics/mesa_acrn/appdomain.te
@@ -2,4 +2,4 @@
 #allow appdomain surfaceflinger_tmpfs:file { read write };
 allow appdomain hal_graphics_allocator_default_tmpfs:file { read write map };
 allow appdomain gpu_device:dir r_dir_perms;
-allow { appdomain -isolated_app } sysfs_app_readable:file r_file_perms;
+allow { appdomain -isolated_app_all } sysfs_app_readable:file r_file_perms;
diff --git a/vendor/domain.te b/vendor/domain.te
index f4fa9c6..017c753 100644
--- a/vendor/domain.te
+++ b/vendor/domain.te
@@ -1 +1 @@
-r_dir_file({ domain -isolated_app }, sysfs_virtio)
+r_dir_file({ domain -isolated_app_all }, sysfs_virtio)
-- 
2.17.1

